### DISCUSSION TITLE
쓰면서 느끼며 개선 하면 좋을 것 같은 점

### DISCUSSION THREAD

**Discussion by DogPam** [green1052/DCRefresher-Reborn#11] [Ideas]
쓰면서 느끼며 개선 하면 좋을 것 같은 점
1. 미리보기 댓글 기능 활성화 하지도 않았는데 사용하다보면 자주 임시 접속 차단당하는건지 몇 분동안 안들어가짐 흰 화면 뜸
2. 자체 설정 내보내기 백업 지원해줬으면 좋겠음 (차단, 메모들 색상 전부) 포맷이나 크롬 지웟다 깔면 다시 처음부터 해야해서
3. 메모 리스트 확인하고 수정 하게 해줬으면 좋겠음 한 번 등록하면 수정이 힘듬
-3.1.메모 리스트 카테고리 생성해서 분류 할 수 있게(예: 분탕어그로, 애옹이, 꾸준글 등으로 따로 메모 계속 하지않고 한 번에 카테고리로 메모 분류 하게)
4. 미리보기 기능 아래 흰색 그라데이션 끄고 킬 수 있는 거 선택 가능하게 해줬으면 좋겠음
5. 툴팁 미리보기 표시 창 크기 가로, 세로 픽셀 단위로 조절 가능하게
6. 나만그런건지 모르겠지만 우클릭 미리보기 창 게시물 글자가 가독성이 떨어져서 잘 안쓰고 글 직접 들어가게됨
-6.1. 아마 제목과 글 내용 구분 선이 없어서, '게시글 배경 블러' 모드시 배경이 회색이라, 폰트 장평이 좁아서, 행간/자간이 좀 더 좁아서, 대댓글이 원본처럼 배경 색 진하기로 구별 없고, 댓글 나누는 가로줄 없어서 이 사항들 세부 조절 켜고 끄기 선택 가능하게 해야할 것 같음
7. 미리보기 창이 화면 중간에 생성되는데 길이가 긴 글도 화면 중앙에 창이 시작해서 불편함, 긴 글이면 화면 맨 위 부터 나와야 할듯
8. 새 게시글 효과 애니메이션 속도를 좀더 빠르게 조절 할 수 있게 
9. 확장 프로그램 자체 설정 메뉴 창 가로길이는 넉넉한데 세로길이 좀 늘려줘 너무 짧음
10. 통피들 색 기본이 파랑이고 해외는 빨강이고 이런데 이런 색들 지정 가능하게
-기본은 그렇게 색 있게하고 나는 전부 회색으로 통일하고 유저 메모만 색 넣어서 메모 유저만 색 구분 하고 싶음
11. 설정 화면 스크롤 맨 위로 드래그 해서 올리면 흰색 그라데이션 때문에 스크롤 바 선택이 안됨
스크롤 바 위 제한을 위에 카테고리 아래까지 제한 두던 카테고리 뒤까지 올라가도 스크롤바 클릭 가능하게 해야할듯
12. 문의나 건의, 오류, 개선점 깃헙으로 하면 깃헙 아이디 없는 사람은 접근성 너무 떨어짐 리뷰에 남길 수 밖에 없음

그리고 웹 스토어 페이지에 이미지 깨짐
개인적으로 기본 폰트 변경 너무 두꺼운데 NanumBarunGothicLight 이 나은듯

보통 건의 사항 리뷰에 쓰는데 길어질 것같아서 회원가입하고 여기에 적음

**Comment by green1052** [comment]
1. 디시인사이드 서버에 요청을 과도하게 보내 차단 당해 발생하는 현상입니다, 
자동 새로고침 시간을 줄이시거나 툴팁 미리보기 설정을 끄시면 해당 현상을 완화할 수 있습니다.

2. 추가 예정입니다.

3. ![image](https://user-images.githubusercontent.com/34261355/211199219-bf79df8b-9a08-45ea-98af-7152550f2297.png)
해당 버블을 누르면 수정창이 열리게 했었는데
기능이 작동되지 않네요 수정하겠습니다.

- 3-1. 딱히 필요 없는듯 합니다.

4. 무엇을 말씀하시는건지 모르겠습니다.
5. 비슷한 요청이 있었으나 컨셉에 맞지 않는거 같아 미디어 숨김 기능으로 대채했습니다. #2, #6
6. 해당 현상이 발상하는 글 주소들을 보내주시면 확인 해보겠습니다.
7. 수정하겠습니다.
8. 수정하겠습니다.
9. 이정도면 충분한거 같습니다.
10. 추가하겠습니다.
11. 수정하겠습니다.
12. 크롬 웹스토어, AMO 리뷰 등으로 오류 제보를 받으면 여러 불편사항 떄문에 깃허브로 받고 있습니다.
좋은 방안을 알려주시면 개선해보겠습니다.

**Reply by green1052** [reply]
3. 크롬, 파이어폭스 둘다 작동하네요

**Reply by DogPam** [reply]
4. 미리보기 기능 아래 흰색 그라데이션 끄고 킬 수 있는 거 선택 가능하게 해줬으면 좋겠음
이건 이전 원본에 게시글 위에 커서 올리면 뜨는  '미리보기 튤팁' 에 있었는데('더 읽을 려면 클릭하세요' 뒤에 흰색 그라데이션 있었음) 여기 오면서 사라진거 같음
그리고 버그인지 우클릭 미리보기 창 위로 스크롤 하면 위 항상 게시글로 바로 이동하는데
아래로 스크롤 내리면 아래 게시물로 이동이 될 때가 있고 안될 때 가 많음

**Reply by green1052** [reply]
확인해보겠습니다.

**Reply by DogPam** [reply]
[Honeycam 2023-01-08 23-07-56.webm](https://user-images.githubusercontent.com/122213767/211200671-2da1b646-8325-4af1-b897-e0a9cd3a793f.webm)

3. 버블 클릭해도 수정 창이 나오지 않음

**Reply by DogPam** [reply]
[Honeycam 2023-01-08 23-09-49.webm](https://user-images.githubusercontent.com/122213767/211200799-914f7267-3dd2-4e5a-9c11-989fada0ba02.webm)

스크롤 해당 현상 아래 스크롤 시 글 이동 불가

**Comment by green1052** [comment]
이미지 깨지는 현상은 저도 인지하고 있습니다만 
따로 설명 이미지 등 다른 이미지로 교체하기 귀찮아 내버려두고 있습니다.

**Comment by green1052** [comment]
#12 에서 진행사항을 보실 수 있습니다, 누락된 항목이 있다면 말씀해주세요

**Comment by DogPam** [comment]
또 하나 발견한건
우클릭 미리보기시 스페이스바로 아래로 스크롤 내리는것 바로 안먹힘
미리보기창을 한 번 클릭해야 스페이스로 아래 스크롤하는거 가능
이걸 게시글 우클릭 미리보기 후 미리보기창 선택 활성화 해야 스페이스바로 아래 스크롤 내리기 가능할듯

**Reply by green1052** [reply]
확인했습니다.

**Comment by green1052** [comment]
#12 말씀해주신 대부분의 오류를 수정했습니다.

미리보기 포커스 문제는 따로 해결법을 찾지 못해 미뤄졌으며
새 게시글 효과는 현재 값도 괜찮아 수정하지 않기로 했습니다.

**Comment by green1052** [comment]
또한 IP 정보 색상은 추후 추가될 예정입니다.

**Comment by green1052** [comment]
[3.7.0](https://github.com/green1052/DCRefresher-Reborn/releases/tag/3.7.0) 릴리즈에 반영됐습니다.

**Reply by DogPam** [reply]
개선 수정 감사합니다

오늘추가로 발견 사항이.
1. 크롬 삭제 후 다시 깔았을 때 발견한점
![r2](https://user-images.githubusercontent.com/122213767/211615984-9298717a-b5a0-421b-aca5-058b8a940a89.PNG)

'이 갤러리 기능을 설정해 보세요' 초록 말풍선이 디시에 처음 들어가면 생기는데
우클릭 
![r1](https://user-images.githubusercontent.com/122213767/211616266-d60d9662-d0b7-4437-8afe-e7cd6fbf5e67.PNG)
미리보기 창 보다 상위에 있어서 글 내용이 가려짐

2. 커서 위에 둘 때 생기는 미리보기 탭의 '더 읽으려면 클릭하세요' 글자가 본문 내용 폰트와 행간의 차이가 적어서 보기 불편함 
![r3](https://user-images.githubusercontent.com/122213767/211617568-b9aaafbd-e332-47bd-8077-f3879c2cbe59.PNG)

'더 읽으려면 클릭하세요' 저 글자만 다른 색 입히거나(회색 등), 가운데 정렬, 볼드, 행간 더 띄우기 처리 해야할듯 

그리고 정작 '더 읽으려면 클릭하세요' 라는 말이 필요한 긴 글에는 안뜨고 짤림
![r4](https://user-images.githubusercontent.com/122213767/211618563-7a1bf8fe-3202-4a2d-8c41-213ae97df127.PNG)
![r5](https://user-images.githubusercontent.com/122213767/211618884-2a893d63-38d4-4b0a-b614-b017e1bcd501.PNG)

**Reply by green1052** [reply]
수정하겠습니다.

**Comment by DogPam** [comment]
![image](https://user-images.githubusercontent.com/122213767/212686912-1053e9cf-6636-4fb7-a4ee-6a6b6f1e5b0a.png)

댓글 도배 차단해도 답글 도배 이런식으로 가로줄 남아 있어서 차단 해도 흔적이 보이고 보기 불편함

**Reply by green1052** [reply]
수정하겠습니다.

**Comment by green1052** [comment]
https://github.com/green1052/DCRefresher-Reborn/issues/12 말씀해주신 대부분의 오류를 수정했습니다.

**Comment by green1052** [comment]
[3.8.0](https://github.com/green1052/DCRefresher-Reborn/releases/tag/3.8.0)에서 반영됐습니다.

**Comment by DogPam** [comment]
1. 12번 사항에 최근 확장프로그램 사이트 안에 지원 카테고리가 이런식으로 생긴 것 같은데 활용할 수 있으면 좋을 듯함 현실적으로 일반 사용자가 깃헙 계정이 있지 않으니, 현재는 게시자의 지원 사이트를 방문하세요로 되어 있음
![image](https://user-images.githubusercontent.com/122213767/215284771-8fa1dac7-800e-4f97-80b6-2d9997cd2b88.png)
그래서 확장프로그램 설정 탭에서도 오류 제보 및 개선 사항 링크를 깃헙과 구글 둘 중 하나 갈 수 있게

2. 우클릭 미리보기 창에서  스크롤해서 위아래로 넘어가는거는 개념글에서는 적용이 글 자체 작성 시간 순번만 되고 개념글만 따로 위 아래로 안넘어가짐

3. 또한 우클릭 미리보기 창 만화 같이 긴 스크롤 내리는 게시물 있을 때 스크롤 살짝만 내리고 나가고 다른 게시물 클릭하면 그 게시물 스크롤한 만큼 내려가 있음 맨 위 부터 안나오고
근데 이게 또 랜덤인게 캐시 방식이 어떤지 잘 모르겟지만
스크롤 안하고 게시글 들어가도 게시글 맨 위 부터 안나오고 게시글 중간부터 나오는 경우도 있고, 정상적으로 아예 맨 위 이미지 로딩 되면서 게시글 맨 위 부터 나오기도 함 
![Honeycam 2023-01-29 04-05-35](https://user-images.githubusercontent.com/122213767/215286365-398ebef2-62db-4a1a-bae2-445b621fa39a.gif)





4.
![Honeycam 2023-01-29 03-56-52](https://user-images.githubusercontent.com/122213767/215286437-9990cde0-e57e-4b5c-a891-b83e160e6a7f.gif)


차단된 아이디들만 나오는 글이면 이렇게 아예 안나오는데 자동으로 댓글 나오기라도하는 페이지로 넘어가게 하던가
차단 된 댓글 있지도 못하게 느끼게 댓글 목록에서 차단된 댓글들 아예 없애서 다시 페이지 번호 재 정렬하던가 해야할듯


5. 설정 저장 코드를 따로 안되면 급하게 임시방편으로
카톡 이미지 캡쳐 처럼 페저장할 수 있게 설정 카테고리들 화면 전체 캡쳐 이미지 다운로드 하게 할 수 있으면 좋을듯
지금도 그냥 설정 페이지를 이미지 캡쳐 한 다음 비교해보면서 일일이 다시 설정 하고 있음 이렇게 라도 간접적으로 백업 지원 하면 좋을듯

**Reply by green1052** [reply]
🤔

**Reply by green1052** [reply]
1. 한번 알아보겟습니다.
2. 기존 제작자 분깨서 해당 기능을 게시글 아이디를 +1 -1 하는 방식으로 구현해서 그럽니다.
3. 고쳤습니다.
4. 구현하기 어려울듯합니다.
5. https://gall.dcinside.com/board/view/?id=idolmaster_new1&no=3099649
해당 방식처럼 설정을 추출할 수 있는 코드를 따로 제공하겠습니다.

**Reply by green1052** [reply]
5. https://github.com/green1052/DCRefresher-Reborn/blob/main/migration.md

**Reply by green1052** [reply]
1. 디스코드 채널을 개설했습니다.
